src/components/Contacts.js

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from "react";
+import Contact from "./Contact";
+import contacts from "./../contacts.json";
+const Contacts = () => {
+  const data = contacts.slice(0, 5);
+
+  const [addContact, setAddContact] = useState(data);
+  const random = Math.floor(Math.random() * contacts.length);
+
+  const generateRandomContact = () => {
+    setAddContact([...addContact, contacts[random]]);
+  };
+
+  //Creamos una nueva data
+  const [sortContact, setSortContact] = useState(addContact);
+  //Funciones de sort
+  const sortByName = () => {
+    //ordenar alfabeticamente DESC
+    setSortContact([
+      sortContact
+        .sort((a, b) => {
+          if (a.name < b.name) return 1;
+          if (a.name > b.name) return -1;
+          if (a.name === 0) return 0;
+        })
+        .reverse(),
+    ]);
+  };
+
+  const sortByPopularity = () => {
+    setSortContact([sortContact.sort((a, b) => b.popularity - a.popularity)]);
```

Comment:
Sort modifica el arreglo es decir no retorna nada entonces, necesitamos hacer una copia y mandarlo por el set sino hacemos esto React no hace el re render, ojo aca tu en tu return tienes el map del primer estado entonces trabajemos con ese 
```suggestion
    setAddContact([...addContact.sort((a, b) => b.popularity - a.popularity)]);
```